internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -102,15 +107,22 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, awsCli
 		clusterResources.WorkerSecurityGroupIDs,
 		allowSSHCIDRS,
 	)
-	if err != nil {
+	// Only release VPC resources, if create cluster request was for the primary cluster.
+	if err != nil && kopsMetadata.ChangeRequest.VPC == "" {
```

Comment:
There is an error in this logic. We no longer are just checking for just `err != nil` which means if `err != nil` and `kopsMetadata.ChangeRequest.VPC != ""` then the code just proceeds as if there was no error.

Reply:
Sorry, forgot to remove the additional VPC condition here, it is not required any more because `release_vpc` handle the resource release for both primary & secondary cluster. 
modify the condition and did a quick cluster/ delete test. looks fine to me.